packages/messenger-widget/src/components/ConfigureProfile/ConfigureProfile.tsx

```diff
@@ -119,16 +123,24 @@ export function ConfigureDM3Profile() {
 
     // changes network on naming service change
     const changeNetwork = (serviceName: string) => {
-        const chainId = fetchChainIdFromServiceName(serviceName);
-        if (chainId && chainId !== chain?.id) {
+        const chainId = fetchChainIdFromServiceName(
+            serviceName,
+            state.dm3Configuration.chainId,
+        );
+        if (chainId && chainId !== chainId) {
```

Comment:
Can this condition ever be true?

Reply:
ohh, same variable names made this confusion. Will fix this